src/StreamsAPI_Lambda/Students/Main.java

```diff
@@ -0,0 +1,148 @@
+package StreamsAPI_Lambda.Students;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.*;
+
+public class Main {
+    public static void main(String[] args) {
+        List <Students> students = new ArrayList<>();
+        students.add(new Students("John", "Doe", 30, "Male", "Mechanical Engineering", 122));
+        students.add(new Students("Jane", "Smith", 22, "Female", "Computer Engineering", 324));
+        students.add(new Students ("Ann", "Gurnmeister", 56, "Female", "Biotech Engineering", 64));
+        students.add(new Students("Elon", "Gated", 27, "Male", "Mechanical Engineering", 90));
+        students.add(new Students("Justin", "Case", 26, "Male", "Computer Engineering", 340));
+        students.add(new Students("Dianne", "Ameter", 31, "Female", "Biotech Engineering", 128));
+        students.add(new Students("Joss", "Sticks", 23, "Male", "Computer Engineering", 20));
+
+        List <Students> filteredStudents = students.stream()
+                .filter(a -> a.getFirstName().startsWith("A"))//Find a list of students whose first name starts with 'A'
+                .collect(toList());
+        System.out.println("\n Name whose first name starts with A");
+        for (Students s : filteredStudents) {
+            System.out.println(s);
+        }
+
+        Map<String, Set<String>> namesByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,
+                        mapping(
+                                Students::getFirstName,
+                                toSet())
+                ));
+        System.out.println("\n Group the students by department names");
+        namesByDepertament.forEach((departamentName,lastName) -> System.out.println(departamentName + " : " + lastName));
+
+        //total students
+        long total = students.stream().count();
+        System.out.println("\n Total number of students: " + total);
+
+        //maAge
+        Comparator<Students> studentsComparator = Comparator.comparing(Students::getAge);
+        Students maxAge = students.stream().max(studentsComparator).get();
+        System.out.println("Max age: " + maxAge);
+
+        //departaments names
+        Map<String, Set<String>> namessByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,
+                        mapping(
+                                Students::getDepartamentName,
+                                toSet())
+                ));
+        System.out.println("\n Group  by department names");
+        namesByDepertament.forEach((departamentName, d) -> System.out.println(departamentName));
```

Comment:
an easier way to do it is `students.stream().map(s -> s.getDepartmantName()).distinct()`